### ISSUE TITLE
What devices does nvbios support?

### ISSUE THREAD

**Issue by alexforencich** [envytools/envytools#116]
It seems to not want to parse Quadro K3000M, K4000M, and K5000M BIOSes.  Am I doing something wrong, or are Kepler cards not supported by nvbios?

**Comment by imirkin** [comment]
Should work. Perhaps you are giving it something that is not the pure BIOS?

**Comment by alexforencich** [comment]
Not sure.  Try the file from this page: https://www.techpowerup.com/vgabios/165689/165689 .  I am seeing output like this:

    $ nvbios Dell\ K5000M\ -\ 80.04.33.00.35.rom
    warning: No strap specified
    Unknown chipset detected: 0
    Unknown DACLOAD table version 5.8
    Failed to parse DACLOAD table at 0x5220 version 5.8
    Unknown IUNK21 table version 0.0
    Failed to parse IUNK21 table at 0x9a91 version 0.0
    Unknown DACLOAD table version 12.3
    Failed to parse DACLOAD table at 0x314b version 12.3
    requested OOB u8 at 0x73726556
    Unknown SENSE table version 0x0
    requested OOB u8 at 0x206e6f69
    Unknown POWER BUDGET table version 0x0
    requested OOB u8 at 0x302e3038
    Unknown BOOST table version 0x0
    requested OOB u8 at 0x33332e34
    Unknown CSTEP table version 0x0
    requested OOB u8 at 0x2e30302e
    BASE CLOCKS table version 0x0
    requested OOB u8 at 0xd203533
    Unknown UNK3C table version 0x0
    requested OOB u8 at 0x6f43000a
    Unknown UNK40 table version 0x0
    requested OOB u8 at 0x69727970
    Unknown UNK44 table version 0x0
    requested OOB u8 at 0x20746867
    Unknown UNK48 table version 0x0
    requested OOB u8 at 0x20294328
    Unknown UNK4c table version 0x0
    Error when parsing mem train: subentries = 8 > 8
    Unknown UNK0D table version 0x66
    Unknown UNK2 table version 0xb8
    Unknown L UNK0 table version 0x14
    Unknown V UNK0 table version 0xe
    Unknown DCB table version 13.10
    Failed to parse DCB table at 0xffff version 13.10
    HWINFO ext sig mismatch [0x00]
    BIOS size 0x15e00 [orig: 0x15e00], 0 valid parts:

    WARN: Couldn't read part 0!

    BIOS type: NV04

    Subsystem id: 0x0000:0x0000

    HWEA at 0x1536c:
        R[0x000000] <= 0x00000000
        R[0x000000] <= 0x00000000
        R[0x1fffffc] &= 0xffffffff |= 0xffffffff

    BIT table at 0x03c0 version 1
    BIT table '2' version 1 at 0x0248 length 0x0004
    BIT table 'B' version 2 at 0x0254 length 0x0021
    BIT table 'C' version 1 at 0x0275 length 0x000e
    BIT table 'D' version 1 at 0x0283 length 0x0004
    BIT table 'A' version 1 at 0x0287 length 0x0003
    BIT table 'I' version 1 at 0x028a length 0x0012
    BIT table 'L' version 1 at 0x029c length 0x0002
    BIT table 'M' version 2 at 0x029e length 0x0011
    BIT table 'N' version 0 at 0x0000 length 0x0000
    BIT table 'P' version 2 at 0x02af length 0x0050
    BIT table 'S' version 2 at 0x02ff length 0x0018
    BIT table 'T' version 1 at 0x0317 length 0x0002
    BIT table 'U' version 1 at 0x0319 length 0x0003
    BIT table 'V' version 1 at 0x031c length 0x0006
    BIT table 'x' version 1 at 0x0322 length 0x0008
    BIT table 'd' version 1 at 0x032a length 0x0002
    BIT table 'p' version 1 at 0x032c length 0x000f
    BIT table 'i' version 2 at 0x033c length 0x0044

    Unknown BIT table 'B' version 2

    Unknown BIT table 'C' version 1

    Unknown BIT table 'I' version 1

    Unknown BIT table 'S' version 2

    Unknown BIT table 'U' version 1

    Unknown BIT table 'V' version 1

    Unknown BIT table 'x' version 1

    Unknown BIT table 'p' version 1

    INFO: BIOS version 0x00.00.00.00.00 from ev    for unknown chipset
    INFO: features (none), unk07 0x00 0x00 0x43 0x68 0x69 0x70, unk19 0x00 0x00 0x00 0x00 0xba 0x91 0x98 0x96, unk23 0x9a, unk24 0x8d, unk25 0x96, unk26 0x9891, unk28 0xdf 0xad 0x9a 0x93 0x9a 0x9e 0x8c 0x9a, unk30 ��߱��߹��߯, unk3c 0x8d 0x90 0x9b 0x8a 0x9c 0x8b 0x96 0x90

All sorts of unknown stuff, zero valid parts, unknown chipset, versions and IDs all read as 0, etc.

**Comment by imirkin** [comment]
Yeah, that's not a pure file, it's probably some sort of formatted thing. Or perhaps it includes more stuff and the vbios is somewhere inside. Extract it using one of the supported methods, and it should work. Notably /sys/kernel/debug/dri/0/vbios.rom if you're running nouveau.

**Comment by Echelon9** [comment]
Yes, techpowerup's VBIOS are imported with some type of header/wrapper.

I recall if you strip off the first 2560 bytes, you will have a VBIOS in the same format that nouveau's supported methods produce as well. It can then be read by `nvbios`
